### PR TITLE
235 include example data

### PR DESCRIPTION
**Pull request by BalzaniEdoardo**
In this PR we address issue #235 and we move osf_download out of tests.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -15,15 +15,15 @@ __pycache__/
...
-data/tid2013
+src/plenoptic/data/plenoptic-test-files
+src/plenoptic/data/ssim_images
+src/plenoptic/data/ssim_analysis.mat
+src/plenoptic/data/msssim_images
+src/plenoptic/data/MAD_results
+src/plenoptic/data/imagenet1000_clsidx_to_labels.txt
+src/plenoptic/data/portilla*
+src/plenoptic/data/FashionMNIST
+src/plenoptic/data/tid2013
```
Comment (BalzaniEdoardo) [inline .gitignore:26]:
```suggestion
data/plenoptic-test-files
data/ssim_images
data/ssim_analysis.mat
data/msssim_images
data/MAD_results
data/imagenet1000_clsidx_to_labels.txt
data/portilla*
data/FashionMNIST
data/tid2013
```

**Review by BalzaniEdoardo — COMMENTED** [review]
revert notebook stuff

**Comment by BalzaniEdoardo** [comment]
@billbrod remember to remove unused figures in src/plenoptic/data.

**Comment by billbrod** [comment]
I've expanded upon Edoardo's changes:
- removed `data/` directory at the root of the repo: we can just download those images (used for tests) from the OSF
- removes my custom functions for downloading data from osf, now uses [pooch](https://www.fatiando.org/pooch/latest/index.html) to do so, which has variety of useful functionality
- standardized how notebooks get images (some were still calling `plt.imread`, which will fail)
- if `torchvision` not installed, we no longer install it in notebook, but raise an error telling people to install it.
- removed unecessary images from `src/plenoptic/data`
- restructured how user interacts with `plenoptic.data`, to look more like scipy's: separate method for each image they can load in.

**Review by billbrod — APPROVED** [review]
Thanks for putting this together Edoardo!

**Pull request merged**